### PR TITLE
Do not report wrapped targets as unused targets

### PR DESCRIPTION
**Pull request by thesmallbang**
Account for wrapped targets when determining if they are unused targets.

Related to issue: https://github.com/NLog/NLog/issues/2288

### CODE REVIEW

File: src/NLog/Targets/Wrappers/DebugTargetWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    using System.ComponentModel;
+    using System.Threading;
+    using NLog.Common;
+    using NLog.Internal;
+
+    /// <summary>
+    /// Wrapper for unit testing with DebugTarget
+    /// </summary>
+    [Target("DebugWrapper", IsWrapper = true)]
+    public class DebugTargetWrapper : WrapperTargetBase
```
Comment (304NotModified) [inline src/NLog/Targets/Wrappers/DebugTargetWrapper.cs]: Please add this one to the test project, thx!

Comment (thesmallbang) [reply src/NLog/Targets/Wrappers/DebugTargetWrapper.cs]: Are you asking for that class to be moved into the testing namespace or for unit tests to be created for the class? I see the DebugTarget exists in the main project. Thanks in advance for the clarification.

Comment (304NotModified) [reply src/NLog/Targets/Wrappers/DebugTargetWrapper.cs]: Move to the test project please, so it's only for the unit tests visible. 

You're right about the debug target. But I prefer to only add public surface with a feature request.

Comment (thesmallbang) [reply src/NLog/Targets/Wrappers/DebugTargetWrapper.cs]: The target being in the test project does not seem to work. I get a target not found. The same thing originally was occurring (and still is) when i tried to use the MockTargetWrapper. It appears that only targets in the main project are being found.

Comment (thesmallbang) [reply src/NLog/Targets/Wrappers/DebugTargetWrapper.cs]: Finished. Adding the assembly reference to unit test project in the nlog config string was enough.

**Comment by thesmallbang** [comment]
Anything pending here? Let me know.

**Comment by 304NotModified** [comment]
Only failing builds, but it looks like that are unstable tests

**Comment by 304NotModified** [comment]
Thanks for the changes! Looks good!

**Comment by 304NotModified** [comment]
merged! Thx! will be released in NLog 4.5

**Pull request merged**